src/pages/index.js

```diff
@@ -18,80 +19,212 @@ import {
+  deleteCardPopup.open();
 }
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
+// Editing Profile //
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  (inputValues) => {
+    editAvatarPopup.renderLoading(true);
+
+    const avatarData = {
+      avatar: inputValues.imageURL,
+    };
+
+    api
+      .editProfileImage(avatarData)
+      .then(() => {
+        userInfoComponent.setAvatar(inputValues.imageURL);
+        editAvatarPopup.close();
+      })
+      .catch((err) => {
+        console.error("Error updating avatar:", err);
+      })
+      .finally(() => {
+        editAvatarPopup.hideLoading();
+      });
   },
-  containerSelector
+  "Save"
 );
 
-section.renderItems();
+profileAvatarButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+});
 
-// Modal Image
-const imagePreviewModal = new PopupWithImage(imageModalSelector);
-imagePreviewModal.setEventListeners();
+editAvatarPopup.setEventListeners();
+
+// Likes///
+function handleLikeClick(card) {
+  const isLiked = card.isLiked();
+  const cardID = card._cardID;
+
+  if (isLiked) {
+    api
+      .likeCountRemove(cardID)
+      .then((updatedCard) => {
+        card.setLikes(updatedCard.likes);
+      })
+      .catch((error) => {
+        console.error("Error removing like:", error);
+      });
+  } else {
+    api
+      .likeCountAdd(cardID)
+      .then((updatedCard) => {
+        card.setLikes(updatedCard.likes);
+      })
+      .catch((error) => {
+        console.error("Error adding like:", error);
+      });
+  }
+}
 
 // Functions
+
+function createCardElement(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleCardImageClick,
+    handleDeleteClick,
+    handleLikeClick,
+    currentUserId
+  );
+
+  return card.getView();
+}
+
 function addCard(data) {
   const cardData = {
     name: data.title,
     link: data.imageURL,
   };
 
-  renderCard(cardData);
-  addCardFormPopup.close();
+  // Show loading state
+
+  addCardFormPopup.renderLoading(true);
+  api
+    .addNewCard(cardData)
+    .then((newCard) => {
+      const cardElement = createCardElement(newCard);
+      section.addItem(cardElement);
+      addCardFormPopup.close();
+    })
+    .catch((error) => {
+      console.error("Error adding card:", error);
+    })
+    .finally(() => {
+      addCardFormPopup.hideLoading();
+    });
 }
+const api = new Api(apiURL);
+
+const imagePreviewModal = new PopupWithImage(imageModalSelector);
+imagePreviewModal.setEventListeners();
+
+function handleCardImageClick(cardData) {
+  imagePreviewModal.open(cardData);
+}
+
+let section;
+
+api
+  .getUserInfo()
+  .then((currentUser) => {
+    currentUserId = currentUser._id;
+
+    userInfoComponent.setUserInfo(currentUser.name, currentUser.about);
+    userInfoComponent.setAvatar(currentUser.avatar);
+
+    // Set the user info on the profile modal
+    profileTitleInput.value = currentUser.name;
```

Comment:
Needs correction: you should fill the form before opening the profile popup, not here